### ISSUE TITLE
`mix credo` fails to run: `no function clause matching in Credo.Code.Name.full/1`

### ISSUE THREAD

**Issue by arubis** [code-corps/code-corps-api#595]
# Problem

A fresh clone of the `develop` branch with passing tests chokes on `mix credo` returning `FunctionClauseError` citing no matching clause on `Credo.Code.Name.full/1`.

@daveconnis pointed out that this is similar in name and behavior to rrrene/credo#242, which is in summary:
> Running credo when a module is referenced through an attribute fails due a lack of match of `Credo.Code.Name.full`

That fix (rrrene/credo#242) is in `master` upstream but not yet on a tagged branch (we're on `0.5.3` which is the most recent). I've pulled in `master` at `963e85e40733de6d370c833af7fefc2d377edd8b` and confirmed that `mix credo` runs correctly.

A full dump of the failing output follows.

```
~/dev/code-corps-api (develop) > mix credo
Checking 185 source files (this might take a while) ...
Error while running Elixir.Credo.Check.Design.AliasUsage on lib/code_corps/stripe_service/stripe_external_account_service.ex
** (EXIT from #PID<0.70.0>) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Credo.Code.Name.full/1
        lib/credo/code/name.ex:29: Credo.Code.Name.full({:@, [line: 8], [{:api, [line: 8], nil}]})
        (elixir) lib/enum.ex:1184: Enum."-map/2-lists^map/1-0-"/2
        lib/credo/code/name.ex:34: Credo.Code.Name.full/1
        lib/credo/code/module.ex:168: Credo.Code.Module.find_dependent_modules/2
        (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
        (elixir) lib/macro.ex:262: Macro.do_traverse/4
        (elixir) lib/macro.ex:268: Macro.do_traverse/4
        (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3

19:04:35.495 [error] Task #PID<0.419.0> started from #PID<0.70.0> terminating
** (FunctionClauseError) no function clause matching in Credo.Code.Name.full/1
    lib/credo/code/name.ex:29: Credo.Code.Name.full({:@, [line: 8], [{:api, [line: 8], nil}]})
    (elixir) lib/enum.ex:1184: Enum."-map/2-lists^map/1-0-"/2
    lib/credo/code/name.ex:34: Credo.Code.Name.full/1
    lib/credo/code/module.ex:168: Credo.Code.Module.find_dependent_modules/2
    (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (elixir) lib/macro.ex:262: Macro.do_traverse/4
    (elixir) lib/macro.ex:268: Macro.do_traverse/4
    (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
Function: #Function<13.6456213/0 in Credo.Check.Runner.run/2>
    Args: []

19:04:35.506 [error] GenServer Credo.Supervisor terminating
** (FunctionClauseError) no function clause matching in Credo.Code.Name.full/1
    lib/credo/code/name.ex:29: Credo.Code.Name.full({:@, [line: 8], [{:api, [line: 8], nil}]})
    (elixir) lib/enum.ex:1184: Enum."-map/2-lists^map/1-0-"/2
    lib/credo/code/name.ex:34: Credo.Code.Name.full/1
    lib/credo/code/module.ex:168: Credo.Code.Module.find_dependent_modules/2
    (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (elixir) lib/macro.ex:262: Macro.do_traverse/4
    (elixir) lib/macro.ex:268: Macro.do_traverse/4
    (elixir) lib/enum.ex:1247: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
Last message: {:EXIT, #PID<0.70.0>, {%FunctionClauseError{arity: 1, function: :full, module: Credo.Code.Name}, [{Credo.Code.Name, :full, [{:@, [line: 8], [{:api, [line: 8], nil}]}], [file: 'lib/credo/code/name.ex', line: 29]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1184]}, {Credo.Code.Name, :full, 1, [file: 'lib/credo/code/name.ex', line: 34]}, {Credo.Code.Module, :find_dependent_modules, 2, [file: 'lib/credo/code/module.ex', line: 168]}, {Enum, :"-map_reduce/3-lists^mapfoldl/2-0-", 3, [file: 'lib/enum.ex', line: 1247]}, {Macro, :do_traverse, 4, [file: 'lib/macro.ex', line: 262]}, {Macro, :do_traverse, 4, [file: 'lib/macro.ex', line: 268]}, {Enum, :"-map_reduce/3-lists^mapfoldl/2-0-", 3, [file: 'lib/enum.ex', line: 1247]}]}}
State: {:state, {:local, Credo.Supervisor}, :one_for_one, [{:child, #PID<0.206.0>, Credo.Service.SourceFileIssues, {Credo.Service.SourceFileIssues, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileIssues]}, {:child, #PID<0.205.0>, Credo.Service.SourceFileScopes, {Credo.Service.SourceFileScopes, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileScopes]}, {:child, #PID<0.204.0>, Credo.Service.SourceFileCodeOnly, {Credo.Service.SourceFileCodeOnly, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileCodeOnly]}, {:child, #PID<0.203.0>, Credo.Service.SourceFileWithoutStringAndSigils, {Credo.Service.SourceFileWithoutStringAndSigils, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileWithoutStringAndSigils]}], :undefined, 3, 5, [], 0, Supervisor.Default, {:ok, {{:one_for_one, 3, 5}, [{Credo.Service.SourceFileWithoutStringAndSigils, {Credo.Service.SourceFileWithoutStringAndSigils, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileWithoutStringAndSigils]}, {Credo.Service.SourceFileCodeOnly, {Credo.Service.SourceFileCodeOnly, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileCodeOnly]}, {Credo.Service.SourceFileScopes, {Credo.Service.SourceFileScopes, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileScopes]}, {Credo.Service.SourceFileIssues, {Credo.Service.SourceFileIssues, :start_link, []}, :permanent, 5000, :worker, [Credo.Service.SourceFileIssues]}]}}}
```

## Subtasks
- [ ] Determine whether to change our code to call `Credo.Code.Name.full` differently or just wait for upstream/master to be released
- [ ] If a local fix, dig out where our code is using the module-referenced-as-attribute calling style, and determine an alternative way to call the same function with different syntax

## References
rrrene/credo#242

Progress on: #

**Comment by joshsmith** [comment]
@arubis thanks for the thorough writeup here! This is indeed an issue on credo's side and the linked issue you have there is marked as closed, but as you confirmed, is only closed on `master`.

Once we see a new version pushed to Hex, we can pull in that version and confirm `mix credo` runs without issue, then close this.

**Comment by joshsmith** [comment]
For the moment, I have just marked this as `blocked`.

**Comment by joshsmith** [comment]
This will be fixed when we move up to `0.6`, which is in `rc` right now.

**Comment by joshsmith** [comment]
Closed by #663.
